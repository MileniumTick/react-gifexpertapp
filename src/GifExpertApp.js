import { useState  } from "react";
import { AddCategory } from "./components/AddCategory";
import { ChangeView } from "./components/ChangeView";
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ( {defaultCategories = ['One Punch']} ) => {

    
    const [categories, setCategories] = useState(defaultCategories);
    
    const [origen, setOrigen] = useState(false);

    return <>
        <h2>GifExpertApp</h2>
        <div className="container">
        <AddCategory setCategorie={ setCategories }/>
        <ChangeView origen={origen} setOrigen={setOrigen}/>
        <br/>
        </div>
        <ol>
            {
                categories.map( category => (
                    <GifGrid key={category} category={category} origen={origen}/>
                ))
                
            }
        </ol>

        
    </>
      

}

