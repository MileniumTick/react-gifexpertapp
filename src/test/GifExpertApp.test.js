import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'


describe('Pruebas en <GifExpertApp />', () => {
    
        const setCategories = jest.fn();
        const setOrigen = jest.fn();
        
        test('Debe de mostrarse correctamente', () => {
            const wrapper = shallow(<GifExpertApp origen={setOrigen} categories={setCategories}/>);
            expect(wrapper).toMatchSnapshot();
        })

        test('Debe de mostrar una lista de categorias', () => {

            const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
            const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

            expect(wrapper).toMatchSnapshot();
            expect(wrapper.find('GifGrid').length).toBe(categories.length);
            
        })
})