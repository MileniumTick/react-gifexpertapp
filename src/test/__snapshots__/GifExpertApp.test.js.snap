// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <div
    className="container"
  >
    <AddCategory
      setCategorie={[Function]}
    />
    <ChangeView
      origen={false}
      setOrigen={[Function]}
    />
    <br />
  </div>
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
      origen={false}
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
      origen={false}
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
      origen={false}
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <div
    className="container"
  >
    <AddCategory
      setCategorie={[Function]}
    />
    <ChangeView
      origen={false}
      setOrigen={[Function]}
    />
    <br />
  </div>
  <ol />
</Fragment>
`;
